components/standings/standings_storage.lua

```diff
@@ -45,4 +46,39 @@ function StandingsStorage.run(index, data)
 	)
 end
 
+function StandingsStorage.fromTemplate(frame)
+	local data = Arguments.getArgs(frame)
+
+	if not data.standingsindex or not data.roundindex or not data.placement then
+		return
+	end
+	if not data.team and not data.player then
+		return
+	end
+
+	if data.team then
+		-- finds [[page|display]] and skips images (images have multiple |)
+		data.team = string.match(data.team, '%[%[([^|]-)|[^|]-%]%]')
+		if data.team and mw.ext.TeamTemplate.teamexists(data.team) then
+			local date = Variables.varDefaultMulti('tournament_startdate', 'tournament_enddate', nil)
+			local team = mw.ext.TeamTemplate.raw(data.team, date)
+			data.participant = team.page
+			data.participantdisplay = team.name
+			data.participantflag = ''
+			data.icon = team.image
+			data.icondark = team.imagedark
+		else
+			data.participant = 'tbd'
+			data.participantdisplay = 'TBD'
+		end
+	elseif data.player then
+		data.participant, data.participantdisplay = string.match(data.team, '%[%[([^|]-)|([^|]-)%]%]')
```

Comment:
should be data.player not data.team
it also would not work for players without custom links as they are displayed as `[[displayName]]` instead of `[[displayName|displayName]]` on some wikis

i think something like the following might work
```suggestion
		-- retrieve flag from input
		local data.participantflag = string.match(data.player, '<span class="flag">%[%[File:[^|]-|([^|]-)|?[^%]]-%]%]</span>')
		-- remove the flag part from the input string
		local playerInput = string.gsub(data.player, '<span class="flag">%[%[File:[^%]]-%]%]</span>', '')
		-- fetch the player display from the input
		local playerDisplay = string.match(playerInput, '%[%[([^%]]-)%]%]')
		if String.isEmpty(playerDisplay) then
			return
		end
		local player = mw.text.split(playerDisplay, '|')
		data.participant = player[1]
		data.participantdisplay = player[2] or player[1]
```
(untested)

Reply:
Can we leave the player support for a follow-up PR? Fixed the typo in the string match

Reply:
sure, just wanted to add what i think should work for most wikis

for sc, sc2, warcraft even what i suggested will not work (due to race display)

Reply:
I'll make an issue for it

Reply:
#1211